com.unity.shadergraph/Editor/Data/Graphs/VirtualTextureShaderProperty.cs

```diff
@@ -100,8 +100,14 @@ internal override void ForeachHLSLProperty(Action<HLSLProperty> action)
 
                 Action<ShaderStringBuilder> customDecl = (builder) =>
                 {
-                    // declare texture stack
+                    //Declare sampler if not defined already, use the first layer to get the proper settings for VT textures.
+                    builder.AppendIndentation();
+                    builder.AppendLine("#ifndef VT_SAMPLER");
+                    builder.AppendLine("#define VT_SAMPLER " + "sampler" + referenceName + "_c0");
+                    builder.AppendLine("SAMPLER(sampler" + referenceName + "_c0);");
```

Comment:
We might be able to declare this sampler as an actual HLSLProperty (see code just above this).

I guess the question is whether it properly de-duplicates identical HLSLProperties, if it is multiply-defined.

What you have here would work, but may break hybrid compatibility if/when they introduce Bind Sets..  but we could also look to fix this code when that happens.

Reply:
It works if I use a custom declaration HLSL property.
https://github.com/Unity-Technologies/Graphics/pull/3538/commits/c564d658571feb1f4b5a5712447224292bf225b9